### ISSUE TITLE
Constant conditions

### ISSUE THREAD

**Issue by kalessil** [kalessil/phpinspectionsea#1207]
`if ($i === false) { ... ($i === false ? 'foo' : 'bar') ... }`

**Comment by orklah** [comment]
Nice!

Should not be limited to ternary

```php
    if($a === true){
        if($a === false){
            
        }
    }
```

Doesn't throw warnings either.

This could also be spotted and simplified:
```php
    if($a === true){
        ...
        if($a === true){
            
        }
    }
```

**Comment by andreasschroth** [comment]
Funny, I just wanted to create an issue with the same suggestion.

Very useful pattern! 👍

**Comment by kalessil** [comment]
Done!
